### ISSUE TITLE
SegF at Aquaria/ScriptInterface.cpp:4719

### ISSUE THREAD

**Issue by tomreyn** [AquariaOSE/Aquaria#12]
While swimming around with my nautilus companion and Li in the Abyss, I ran into a SegF on this (fully patched) Ubuntu 14.04 x86_64 system, running my own Debug build of the experimental branch (HEAD as of today). 

A core file was written (due to ```ulimit -c unlimited```), which I then analyzed:

```
tomreyn@atibox:~/aquaria$ gdb -q -n -ex "bt" -batch aquaria.debug-dynamic core

warning: core file may not match specified executable file.
[New LWP 19506]
[New LWP 19511]
[New LWP 19510]
[New LWP 19509]
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
Core was generated by `./aquaria.debug-dynamic'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  l_entity_damage (L=0x3d76e00) at /home/tomreyn/SCM/aquaria/Aquaria/ScriptInterface.cpp:4719
4719			didDamage = e->damage(d);
#0  l_entity_damage (L=0x3d76e00) at /home/tomreyn/SCM/aquaria/Aquaria/ScriptInterface.cpp:4719
#1  0x00007fb9bf3d6f78 in luaD_precall (L=L@entry=0x3d76e00, func=0x3df90c0, nresults=nresults@entry=0) at ldo.c:320
#2  0x00007fb9bf3e16af in luaV_execute (L=L@entry=0x3d76e00, nexeccalls=nexeccalls@entry=1) at lvm.c:591
#3  0x00007fb9bf3d73cd in luaD_call (L=0x3d76e00, func=0x3df9090, nResults=<optimized out>) at ldo.c:378
#4  0x00007fb9bf3d66eb in luaD_rawrunprotected (L=L@entry=0x3d76e00, f=f@entry=0x7fb9bf3d2050 <f_call>, ud=ud@entry=0x7fffc4486f40) at ldo.c:116
#5  0x00007fb9bf3d755a in luaD_pcall (L=L@entry=0x3d76e00, func=func@entry=0x7fb9bf3d2050 <f_call>, u=u@entry=0x7fffc4486f40, old_top=32, ef=<optimized out>) at ldo.c:464
#6  0x00007fb9bf3d334d in lua_pcall (L=0x3d76e00, nargs=2, nresults=0, errfunc=<optimized out>) at lapi.c:821
#7  0x0000000000548eff in Script::doCall (this=0x318b970, nparams=nparams@entry=2, nrets=nrets@entry=0) at /home/tomreyn/SCM/aquaria/Aquaria/ScriptInterface.cpp:11287
#8  0x000000000054910d in Script::call (this=<optimized out>, name=name@entry=0x644d38 "update", param1=param1@entry=0x3dae2e0, param2=0, param2@entry=0,0333333276) at /home/tomreyn/SCM/aquaria/Aquaria/ScriptInterface.cpp:11337
#9  0x0000000000535297 in ScriptedEntity::onUpdate (this=0x3dae2e0, dt=0,0333333276) at /home/tomreyn/SCM/aquaria/Aquaria/ScriptedEntity.cpp:520
#10 0x00000000005cae26 in RenderObject::update (this=this@entry=0x3dae2e0, dt=dt@entry=0,0333333276) at /home/tomreyn/SCM/aquaria/BBGE/RenderObject.cpp:1116
#11 0x000000000049c1a3 in Entity::update (this=0x3dae2e0, dt=0,0333333276) at /home/tomreyn/SCM/aquaria/Aquaria/Entity.cpp:1173
#12 0x000000000059e9b6 in Core::updateRenderObjects (this=this@entry=0x7fffc4487260, dt=0,0333333276) at /home/tomreyn/SCM/aquaria/BBGE/Core.cpp:2776
#13 0x000000000040b972 in Core::main (this=this@entry=0x7fffc4487260, runTime=runTime@entry=-1) at /home/tomreyn/SCM/aquaria/BBGE/Core.cpp:3101
#14 0x0000000000409cc6 in main (argc=<optimized out>, argv=<optimized out>) at /home/tomreyn/SCM/aquaria/Aquaria/Main.cpp:113
tomreyn@atibox:~/aquaria$ 
```

**Comment by tomreyn** [comment]
I believe this was fixed by b4906a384044dbac0d2fbf6d6ebe1d9dc9f98228 - thanks!

**Comment by tomreyn** [comment]
I believe this was fixed by b4906a384044dbac0d2fbf6d6ebe1d9dc9f98228 - thanks!
